### PR TITLE
feat(arc): support RGB565A8 arc image

### LINKED ISSUE #5582: V9 two minor issues
### LVGL version

V9.0

### What happened?

Problems found in V9:
1, lv_anim_path_bounce The result returned is abnormal.
2, lv_obj_set_style_arc_image_src(arc, &img_xx, LV_PART_MAIN);
lv_obj_set_style_arc_image_src(arc, &img_xxx, LV_PART_INDICATOR);
ARC can't add a background image.

### How to reproduce?

_No response_

### PR DESCRIPTION
**Pull request by kisvegabor**
fixes #5582

### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/draw/sw/lv_draw_sw_arc.c [hunk 1]
```diff
@@ -45,6 +45,7 @@ static void get_rounded_area(int16_t angle, int32_t radius, uint8_t thickness, l
 /**********************
  *   GLOBAL FUNCTIONS
  **********************/
+#include <stdio.h>
```
Comment (XuNeo) [inline src/draw/sw/lv_draw_sw_arc.c]: Need to remove.

File: src/draw/sw/lv_draw_sw_arc.c [hunk 2]
```diff
@@ -185,6 +198,21 @@ void lv_draw_sw_arc(lv_draw_unit_t * draw_unit, const lv_draw_arc_dsc_t * dsc, c
...
 
+        /*If it was an RGB565A8 image use consider its A8 part on the mask*/
+        if(img_mask && blend_dsc.mask_res != LV_DRAW_SW_MASK_RES_TRANSP) {
+            const uint8_t * img_mask_tmp = img_mask;
+            img_mask_tmp += blend_dsc.src_stride / 2 * (blend_area.y1 - blend_dsc.src_area->y1);
+            img_mask_tmp += blend_area.x1 - blend_dsc.src_area->x1;
+
+            int32_t i;
+            for(i = 0; i < blend_w; i++) {
+                mask_buf[i] = (uint32_t)((uint32_t)mask_buf[i] * img_mask_tmp[i]) >> 8;
```
Comment (XuNeo) [inline src/draw/sw/lv_draw_sw_arc.c]:
```suggestion
                mask_buf[i] = LV_OPA_MIX2(mask_buf[i], img_mask_tmp[i]));
```

File: src/misc/lv_log.c [hunk 3]
```diff
@@ -101,6 +101,7 @@ void _lv_log_add(lv_log_level_t level, const char * file, int line, const char *
                lvl_prefix[level], LOG_TIMESTAMP_EXPR func);
         vprintf(format, args);
         printf(LOG_FILE_LINE_FMT "\n" LOG_FILE_LINE_EXPR);
+        fflush(stdout);
```
Comment (XuNeo) [inline src/misc/lv_log.c:104]: Could you please clarify if this is an intentional implementation or an inadvertent error?

File: src/draw/sw/lv_draw_sw_arc.c (hunk 2 above)
Comment (kisvegabor) [reply src/draw/sw/lv_draw_sw_arc.c]: Updated.

File: src/draw/sw/lv_draw_sw_arc.c (hunk 1 above)
Comment (kisvegabor) [reply src/draw/sw/lv_draw_sw_arc.c]: Removed

File: src/misc/lv_log.c (hunk 3 above)
Comment (kisvegabor) [reply src/misc/lv_log.c:104]: I was fighting a little to display the logs in the failing tests, but I don't remember if it was really required in the end. Anyway, I think it's useful to be sure the logs are always flushed.

**Review by XuNeo — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**